src/Methods/EloquentBuilderForwardsCallsExtension.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of Larastan.
+ *
+ * (c) Nuno Maduro <enunomaduro@gmail.com>
+ *
+ *  For the full copyright and license information, please view the LICENSE
+ *  file that was distributed with this source code.
+ */
+
+namespace NunoMaduro\Larastan\Methods;
+
+use PHPStan\Type\StringType;
+use PHPStan\Type\CallableType;
+use NunoMaduro\Larastan\Concerns;
+use Illuminate\Database\Query\Builder;
+use PHPStan\Reflection\ClassReflection;
+use PHPStan\Reflection\MethodReflection;
+use PHPStan\Reflection\PassedByReference;
+use PHPStan\Reflection\BrokerAwareExtension;
+use PHPStan\Reflection\ParametersAcceptorSelector;
+use PHPStan\Reflection\Dummy\DummyMethodReflection;
+use PHPStan\Reflection\MethodsClassReflectionExtension;
+use PHPStan\Reflection\Native\NativeParameterReflection;
+use Illuminate\Database\Eloquent\Builder as EloquentBuilder;
+use NunoMaduro\Larastan\Reflection\EloquentBuilderMethodReflection;
+
+final class EloquentBuilderForwardsCallsExtension implements MethodsClassReflectionExtension, BrokerAwareExtension
+{
+    use Concerns\HasBroker;
+
+    /**
+     * The methods that should be returned from query builder.
+     *
+     * @var array
+     */
+    protected $passthru = [
+        'insert', 'insertOrIgnore', 'insertGetId', 'insertUsing', 'getBindings', 'toSql', 'dump', 'dd',
+        'exists', 'doesntExist', 'count', 'min', 'max', 'avg', 'average', 'sum', 'getConnection',
+    ];
+
+    /**
+     * @return ClassReflection
+     * @throws \PHPStan\Broker\ClassNotFoundException
+     */
+    private function getBuilderReflection(): ClassReflection
+    {
+        return $this->broker->getClass(Builder::class);
+    }
+
+    public function hasMethod(ClassReflection $classReflection, string $methodName): bool
+    {
+        if ($classReflection->getName() !== EloquentBuilder::class) {
+            return false;
+        }
+
+        if ($methodName === 'macro') {
+            return true;
+        }
+
+        if (in_array($methodName, $this->passthru)) {
+            return true;
+        }
+
+        return true;
```

Comment:
@canvural , why do you have the same action in both branches of the if statement on line 64?

Reply:
This method tries to mimic the magic method `__call` in the Eloquent Builder class. `if` checks are because of that. And the last `return true` is because model scopes can be called on the builder, and they can have any name. So we need to return true always.

Of course, we can just return true, and not do the checks. But in my opinion this reads better, and close to the `__call` in the builder class.

Reply:
Fair enough, @canvural - maybe add your reply as a comment in that code?